packages/react-aria-components/src/Dialog.tsx

```diff
@@ -39,7 +41,9 @@ export const OverlayTriggerStateContext = createContext<OverlayTriggerState | nu
  * A DialogTrigger opens a dialog when a trigger element is pressed.
  */
 export function DialogTrigger(props: DialogTriggerProps) {
-  let state = useOverlayTriggerState(props);
+  // Use useMenuTriggerState instead of useOverlayTriggerState in case a menu is embedded in the dialog.
+  // This is needed to handle submenus.
+  let state = useMenuTriggerState(props);
```

Comment:
Not super ideal...

Reply:
I'm assuming we do this because submenus behave differently than say a picker inside a popover when it comes to closing? So we can't push it down in a generalized overlaytrigger?

Will this be an issue if we have a Dialog with a Menu inside it? will try to answer myself after review

Reply:
> Will this be an issue if we have a Dialog with a Menu inside it? will try to answer myself after review

Theoretically should be fine I imagine since the MenuTrigger in the Menu provides its own menu state context and its wrapped menu with use that rather than the DialogTrigger's